### DISCUSSION TITLE
How to mark transformation a possible to chache

### DISCUSSION THREAD

**Discussion by jakubMitura14** [Project-MONAI/MONAI#5435] [Q&A] [answered]
Hello I am using custom transform that extend MapTransform and smart cache dataset
The default behaviour is not to chache transform (and I fully agree with such decision !) Hovewer when I know that the ransform is not random I would like to mark it so the Cache dataset will understand it - How to achieve it ?

**Comment by KumoLiu** [comment, marked as answer]
Hi @jakubMitura14, thanks for your interest! If your custom transform is randomized, just inherit the `Randomizable`. The `CacheDataset` will cache the transform if the transform is not randomized.
https://github.com/Project-MONAI/MONAI/blob/9eb513a8bbf604cd710f1edef9d98990109fcae9/monai/data/dataset.py#L842-L845

Hope it can help you, thanks!

**Reply by jakubMitura14** [reply]
thanks!
